src/ScottPlot5/ScottPlot/DataSource/XYArrays.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace ScottPlot.DataSource;
+
+/// <summary>
+/// This data source has Xs and Ys defined in fixed-length arrays.
+/// Changes made to the contents of the arrays will appear when the plot is rendered.
+/// </summary>
+public class XYArrays : IDataSource
+{
+    private readonly double[] Xs;
+    private readonly double[] Ys;
+
+    public int Count => Xs.Length;
+
+    public XYArrays(double[] xs, double[] ys)
```

Comment:
Should probably also have a constructor that takes in a `Coordinate[]`

Reply:
I think this would require copying values into Xs and Ys, at which point the link is broken for users to modify values in the original arrays.

Probably the fix here is a `CoordinateArray` class